File: src/Java/507.PerfectNumber.java

import java.awt.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.LinkedList;
import java.util.Queue;
import java.util.HashMap;

class Solution {
    public boolean checkPerfectNumber(int num) {
        if (num <= 1) {
            return false;
        }
        int up = (int)Math.sqrt(num) + 1;
        int sum = 1;
        for (int i = 2; i < up; i++) {
            if (num % i == 0) {
                sum += i;
                if (num / i != i) {
                    sum += num / i;
                }
            }
        }
        return sum == num;
    }
}

File: src/Python/501-600/507.PerfectNumber.py

class Solution:
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        return num in [6,28,496,8128,33550336,8589869056,137438691328]
   
if __name__ == '__main__':
    solution = Solution()
    print(solution.checkPerfectNumber(1))
    print(solution.checkPerfectNumber(28))
else:
    pass
